content/cpp/concepts/operator-overloading/operator-overloading.md

```diff
@@ -0,0 +1,111 @@
+---
+Title: 'Operator Overloading'
+Description: 'Operator overloading redefines operators for user-defined classes.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Operators'
+  - 'Binary'
+  - 'Logical'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+**Operator overloading** redefines operators for user-defined classes. When an overloaded operator is called, the compiler determines which definition to use based on the arguments provided. The following operators can be overloaded:
+
+| ->      | ->\* | new    | new [] | delete | delete []     |
+| ------- | ---- | ------ | ------ | ------ | ------------- |
+| +       | -    | \*     | /      | %      | ^             |
+| +=      | -=   | \*=    | /=     | %=     | ^=            |
+| &#124;= | &    | &#124; | ~      | !      | ,             |
+| =       | &=   | <<=    | >>=    | []     | ()            |
+| <       | >    | <=     | >=     | ++     | --            |
+| <<      | >>   | ==     | !=     | &&     | &#124; &#124; |
+
+The operators below cannot be overloaded:
+
+| ::  | .\* | .   | ?   |
+| --- | --- | --- | --- |
+
+## Syntax
+
+The `operator` keyword is used with the following syntax to overload operators:
+
+```pseudo
+class className {
+
+    public:
+        returnType operator symbol (arguments) {
+        }
+};
+```
+
+- `className`: name of the class
+- `returnType`: return type of the function
+- `symbol`: operator is which to overload
+- `arguments`: parameters to pass in
+
+## Codebyte Example
+
+The following codebyte example overloads the plus (`+`) operator within the `Pyramid` class and returns the volume of two pyramids added together:
+
+```codebyte/cpp
+#include <iostream>
+using namespace std;
+
+class Pyramid{
+  public:
+    // Defining getters and setters
+    double getVolume(void){
+      return (base*height)/3;
+    }
+
+    void setBase(double b){
+      base = b;
+    }
+
+    void setHeight(double h){
+      height = h;
+    }
+
+    // Overload + operator
+    Pyramid operator+(const Pyramid& p){
```

Comment:
Is there supposed to be no space between the operator and everything else? There were spaces in the syntax example.

Reply:
@SSwiniarski I'm not entirely sure what the typical convention is for this. It works with the spacing too. I was taught the way I wrote it here in the example, with no spacing. It is also written with no spacing in Bjarne Stroustrop's "The C++ Programming Language". I do think the pseudo syntax is easier to read with spacing though. Let me know what you think.

Reply:
@KTom101 I agree that the syntax parses better with the spaces. As long as it works with the spaces as well, then I don't think there's a real issue.